core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -2265,12 +2265,35 @@ private void assertTypeAndToString(
     checkSimplify(coalesce(gt(nullInt, nullInt), trueLiteral),
         "true");
     checkSimplify(coalesce(unaryPlus(nullInt), unaryPlus(vInt())),
-        "+(?0.int0)");
-    checkSimplifyUnchanged(coalesce(unaryPlus(vInt(1)), unaryPlus(vInt())));
+        "?0.int0");
+    checkSimplifyUnchanged(coalesce(vInt(1), vInt()));
 
     checkSimplify(coalesce(nullInt, vInt()), "?0.int0");
     checkSimplify(coalesce(vInt(), nullInt, vInt(1)),
         "COALESCE(?0.int0, ?0.int1)");
+
+    checkSimplify(coalesce(vDecimalNotNull(), vDecimal()), // first arg not null
+        "?0.notNullDecimal0");
+    checkSimplifyUnchanged(coalesce(vDecimal(), vDecimalNotNull()));
+    checkSimplify(coalesce(vDecimal(), vDecimal()), // repeated arg
+        "?0.decimal0");
+    checkSimplify(coalesce(vDecimalNotNull(), vDecimalNotNull()), // repeated arg
+        "?0.notNullDecimal0");
```

Comment:
Could you please put the comment on the top? Also squash commits.

Reply:
Done. Thank you.